src/components/ec-metroline-item/ec-metroline-item.spec.js

```diff
@@ -0,0 +1,109 @@
+import { mount } from '@vue/test-utils';
+import { withMockedConsole } from '../../../tests/utils/console';
+import EcMetrolineItem from './ec-metroline-item.vue';
+import * as STATUS from '../../enums/metroline-status';
+
+function mountEcMetrolineItem(props, mountOpts) {
+  return mount(EcMetrolineItem, {
+    propsData: {
+      index: '1',
+      ...props,
+    },
+    ...mountOpts,
+  });
+}
+
+const slots = {
+  heading: '<span>Payee information</span>',
+  'sub-heading': '<span>Amount fully allocated</span>',
+  'header-cta': '<a href="#>Edit</a>',
+  main: '<p> Lorem ipsum dolor sit amet, voluptates at cumque repudiandae atque quod voluptatum, aperiam dignissimos, vitae, neque mollitia repellat! </p>',
+  'footer-cta': '<a href="#>Continue</a>',
+};
+
+describe('EcMetrolineItem', () => {
+  it('should throw an error when we pass a non valid status', () => {
+    withMockedConsole((errorSpy) => {
+      mountEcMetrolineItem({ status: 'a non valid status' });
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "status"');
+    });
+  });
+
+  it('it should render properly when status is "next"', () => {
+    const wrapper = mountEcMetrolineItem(
+      {
+        status: STATUS.NEXT,
+      },
+      {
+        slots,
+      },
+    );
+    expect(wrapper.findByDataTest('ec-metroline-item__status-disk-enumeration').exists()).toBe(true);
+    expect(wrapper.findByDataTest('ec-metroline-item__status-disk-completed').exists()).toBe(false);
+    expect(wrapper.findByDataTest('ec-metroline-item__header-heading').exists()).toBe(true);
+    expect(wrapper.findByDataTest('ec-metroline-item__header-sub-heading').exists()).toBe(false);
+    expect(wrapper.findByDataTest('ec-metroline-item__status-bar').exists()).toBe(true);
+    expect(wrapper.findByDataTest('ec-metroline-item__header-cta').exists()).toBe(false);
+    expect(wrapper.findByDataTest('ec-metroline-item__main').exists()).toBe(false);
+    expect(wrapper.findByDataTest('ec-metroline-item__footer').exists()).toBe(false);
```

Comment:
You are testing all of those in .toMatchSnapshot() call.

Reply:
A lot of thing vary. I would prefer to keep this. Snapshots are useful as well as we can see the slots and classes are ok.